### PR TITLE
Adding module website_twitter_js_self_hosted

### PR DESCRIPTION
**Pull request by gfcapalbo**

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Maybe this one can fit better in OCA/social? Why the no_ext_links part?

**Comment by hbrunn** [comment]
check runbot and travis

File: website_twitter_no_ext_links/__openerp__.py
Comment (hbrunn) [inline website_twitter_no_ext_links/__openerp__.py]: remove empty key or write a summary

File: website_twitter_no_ext_links/data/themes.xml
Comment (hbrunn) [inline website_twitter_no_ext_links/data/themes.xml]: fix indentation

File: website_twitter_no_ext_links/models/twitter_feed.py
Comment (hbrunn) [inline website_twitter_no_ext_links/models/twitter_feed.py]: this makes no sense in combination with a `@api.model` function

File: website_twitter_no_ext_links/models/twitter_feed.py
Comment (hbrunn) [inline website_twitter_no_ext_links/models/twitter_feed.py]: this allows every logged in user to delete all tweets. I don't think we want that

File: website_twitter_no_ext_links/README.rst
Comment (hbrunn) [inline website_twitter_no_ext_links/README.rst]: find out if we're allowed to use the twitter logo in the first place, and if so, name your source here

**Review by hbrunn — CHANGES_REQUESTED** [review]

**Comment by hbrunn** [comment]
@pedrobaeza the reason we wrote this module is that we want our websites to be free of any js that pulls third party code from the net. This as well as privacy measure for the website users as as security measure in case one of those services ever is compromised. As such, it has less to do with social stuff but with the website

**Comment by pedrobaeza** [comment]
Thanks for the reasoning, Holger. Indeed it's a strong argument for the module. About the host in OCA/social, I still see no problem on moving to there, as the main topic is about social aspects. It doesn't mind that it happens on website or backend (mails and so on). See for example https://github.com/OCA/event/tree/8.0/website_event_contact, that is something for the website, but the transversal topic about event makes OCA/event a better hosting.

Anyways, we can hear another opinion on this if you want.

About the name, there are 2 objections:
- The plural on the name, which we at OCA avoids: https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#modules
- It's not very self-descriptive. What about using as name `website_twitter_internal_js` or `website_twitter_js_self_hosted`?

**Comment by gfcapalbo** [comment]
@pedro, confirmed the absence of external js or other links. I do believe it belongs in website, afterall users look for snippets in /website mostly.

I am open to a name change, but  I think website_twitter_no_ext_links  could be ok (the plural!) if i add a summary/readme  description where i specify No_ext_links meaning better.

**Comment by hbrunn** [comment]
``` python

class Pedro(reviewbot):
    def review(self, mr):
        if any(mod.name.endswith('s') for mod in mr.modules):
            self.comment('No plurals!')
        # do review
```

;-)

**Comment by gfcapalbo** [comment]
no  plurals!
so can we agree to improve comprehension of this module by:

0) leaving in website (together with most a snippets)
1) renaming from website_twitter_no_ext_links   to website_twitter_no_ext_link  (no plural)
2) adding extra information in README explanation to describe what no_ext_link means.

?

**Comment by pedrobaeza** [comment]
@hbrunn :laughing: 

@gfcapalbo I won't block this anymore, so if you both decide that this is a good name and repository, go ahead.

**Comment by hbrunn** [comment]
I leave the decision to @gfcapalbo

**Comment by gfcapalbo** [comment]
As you both leave me the decision. I decide; I am thinking no_ext_links is confusing.
So i rename to website_twitter_js_self_hosted  (more clear) ,  add an extra line of explanation in README better and leave it in website repo.
i hope you like my decision, but you abdicated your vote.

**Comment by pedrobaeza** [comment]
:clap: :clap:

File: website_twitter_js_self_hosted/README.rst
Comment (pedrobaeza) [inline website_twitter_js_self_hosted/README.rst:9]: s/twitter/Twitter

File: website_twitter_js_self_hosted/README.rst
Comment (pedrobaeza) [inline website_twitter_js_self_hosted/README.rst:6]: s/javascript/JavaScript

File: website_twitter_js_self_hosted/README.rst
Comment (pedrobaeza) [inline website_twitter_js_self_hosted/README.rst:11]: s/js/JS

Comment (pedrobaeza) [reply website_twitter_js_self_hosted/README.rst:11]: s/form/from

Comment (pedrobaeza) [reply website_twitter_js_self_hosted/README.rst:11]: Don't indent this, as the list looks weird in this case.

File: website_twitter_js_self_hosted/README.rst
Comment (pedrobaeza) [inline website_twitter_js_self_hosted/README.rst:25]: It should be `#.`

File: website_twitter_js_self_hosted/README.rst
Comment (pedrobaeza) [inline website_twitter_js_self_hosted/README.rst:26]: add 2 indentation spaces

File: website_twitter_js_self_hosted/README.rst
Comment (pedrobaeza) [inline website_twitter_js_self_hosted/README.rst:27]: 2 indentation spaces

File: website_twitter_js_self_hosted/README.rst
Comment (pedrobaeza) [inline website_twitter_js_self_hosted/README.rst:28]: 2 indentation spaces

File: website_twitter_js_self_hosted/README.rst
Comment (pedrobaeza) [inline website_twitter_js_self_hosted/README.rst:32]: `#.`

File: website_twitter_js_self_hosted/README.rst
Comment (pedrobaeza) [inline website_twitter_js_self_hosted/README.rst:33]: 2 indentation spaces

File: website_twitter_js_self_hosted/README.rst
Comment (pedrobaeza) [inline website_twitter_js_self_hosted/README.rst:51]: Don't put this

File: website_twitter_js_self_hosted/README.rst
Comment (pedrobaeza) [inline website_twitter_js_self_hosted/README.rst:52]: Put bullets

File: website_twitter_js_self_hosted/README.rst
Comment (pedrobaeza) [inline website_twitter_js_self_hosted/README.rst:64]: Use new template for this.

File: website_twitter_js_self_hosted/__openerp__.py
Comment (pedrobaeza) [inline website_twitter_js_self_hosted/__openerp__.py]: s/javascript/JavaScript

File: website_twitter_js_self_hosted/__openerp__.py
Comment (pedrobaeza) [inline website_twitter_js_self_hosted/__openerp__.py:14]: Put final comma

File: website_twitter_js_self_hosted/data/themes.xml
Comment (pedrobaeza) [inline website_twitter_js_self_hosted/data/themes.xml:4]: Align correctly this

File: website_twitter_js_self_hosted/models/twitter_theme.py
Comment (pedrobaeza) [inline website_twitter_js_self_hosted/models/twitter_theme.py:16]: Isn't this list in any place that we need to recreate it?

**Pull request closed without merging**
